content/articles/online-machine-learning-with-river-python/index.md

```diff
@@ -0,0 +1,446 @@
+
+```python
+def add_all_attributes(package):
+    subpackages = []
+    submodules = []
+    for i in dir(package):
+        if str(i) not in ["__all__", "__builtins__", "__cached__", "__doc__", "__file__", "__loader__", "__name__", "__package__", "__path__", "__pdoc__", "__spec__", "__version__"]:
+            subpackages.append(i)
+            res = [j for j in dir(eval("river.{}".format(i)))]
+            submodules.append(res)
+    df = pd.DataFrame(submodules)
+    df = df.T
+    df.columns = subpackages
+    res_df = df.dropna()
+    return res_df
+```
+
+We name this function `add_all_attributes` to get all the attributes and methods needed to build our model.
+
+The functios loops through the package and remove the followig: `__all__`, `__builtins__`, `__cached__`, `__doc__`, `__file__`, `__loader__`, `__name__`, `__package__`, `__path__`, `__pdoc__`, `__spec__` and `__version__`.
+
+We remove them since are not used to build our model.
+
+The remaining sub-packages and sub-modules in the list are added using the `subpackages.append()` method.
+
+We then add these sub-packages and submodules into our data frame using the `pd.DataFrame(submodules)` method. The data frame is what will be used to train our model. The data frame will now have all the sub-packages and submodules.
+
+To now get these methods and attributes use this command.
+
+```python
+river_df = add_all_attributes(river)
+```
+
+### Simulating streaming data
+
+To use River Python we need streaming data.
+
+Streaming data comes in incrementally over time and comes one at a time. To simulate streaming data we use our dataset as a list of a tuple as shown.
+
+We shall have two lists of data: A train list and a test list.
+
+#### Train list
+
+```python
+data = [("my python program is runnning","software"),
+("I tried to run this program, but it has bugs","software"),
+("I need a new machine","hardware"),
+("the flashdisk is broken","hardware"),
+("We need to test our code","software"),
+("programming concepts and testing","software"),
+("Electrical device","hardware"),
+("device drives","hardware"),
+("The generator is broken","hardware"),
+("im buidling a REST API","software"),
+("design the best API so far","software"),
+("they need more electrical wiring","hardware"),
+("my code has errors","software"),
+("i found some program test faulty","software"),
+("i broke the car handle","hardware"),
+("i tested the user interface code","software")]
+```
+
+This contains a list of texts that are labeled as either `hardware` or `software` related which is used to train the model one at a time.
+
+#### Test List
+
+```python
+test_data = [('he writes programs daily','software'),
+             ('my disk is broken','hardware'),
+             ("program mantainance","software"),
+             ('The drive is full','hardware')]
+```
+
+Used to test our model to gauge how it learned.
+
```

Comment:
Do not start sentences halfway. "This test data/test list will be used....."

That's how your sentences should start. Scan through the whole tutorial and correct where necessary